rslib/src/dbcheck.rs

```diff
@@ -294,6 +295,18 @@ impl Collection {
             }
         }
 
+        let new_tags = self.storage.all_tags_sorted()?;
+        for old in old_tags.into_iter() {
+            for new in new_tags.iter() {
+                if new.name == old.name {
+                    self.storage.set_tag_collapsed(&new.name, new.collapsed)?;
+                    break;
+                } else if new.name.starts_with(&old.name) {
+                    self.set_tag_collapsed(&old.name, old.collapsed)?;
+                }
+            }
+        }
+
```

Comment:
This code would be better split up into a separate function implemented in tags.rs.